cruise-control-client/cruisecontrolclient/client/cccli.py

```diff
@@ -301,26 +141,67 @@ def query_cruise_control(args):
         for parameter in parameters_to_remove:
             endpoint.remove_param(parameter)
 
-    # Handle instantiating the correct URL for the Responder
-    url = generate_url_from_cc_socket_address(args.socket_address, endpoint)
+    return endpoint
 
+
+def get_responder(endpoint: Endpoint.AbstractEndpoint,
+                  url: str) -> AbstractResponder:
     # Handle instantiating the correct Responder
     if endpoint.http_method == "GET":
         json_responder = JSONDisplayingResponderGet(url)
     elif endpoint.http_method == "POST":
         json_responder = JSONDisplayingResponderPost(url)
     else:
-        raise ValueError("Unexpected http_method {} in endpoint".format(endpoint.http_method))
+        raise ValueError(f"Unexpected http_method {endpoint.http_method} in endpoint")
```

Comment:
In case relevant for the other parts of the client (not for the `get_responder`), CC supports `HTTP OPTIONS` method, as well (see the [servlet code](https://github.com/linkedin/cruise-control/blob/master/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServlet.java#L141)).

Reply:
It is relevant; `ccdo` does not yet provide `cors` support